File: doc-testing/src/sdk_openrouter.ts
import "dotenv/config";
import { init, flush, shutdown } from "neatlogs";
import { wrapOpenRouterAgent } from "neatlogs/openrouter-agent";
import { OpenRouter } from "@openrouter/agent";

async function main() {
  await init({
    apiKey: process.env.NEATLOGS_API_KEY,
    workflowName: "openrouter-demo",
  });

  const openrouter = wrapOpenRouterAgent(
    new OpenRouter({ apiKey: process.env.OPENROUTER_API_KEY! }),
  );

  const result = openrouter.callModel({
    model: "openai/gpt-4o-mini",
    input: "In one sentence, what is OpenRouter?",
    temperature: 0.3,
    topP: 0.9,
    maxOutputTokens: 256,
  });
  console.log(await result.getText());

  await flush();
  await shutdown();
}

main().catch(console.error);

File: agents/src/agents/sdk_openrouter.py
import os
from pathlib import Path

from dotenv import load_dotenv
import neatlogs
from openrouter import OpenRouter

load_dotenv(Path(__file__).resolve().parents[2] / ".env")

neatlogs.init(api_key=os.environ["NEATLOGS_API_KEY"], workflow_name="openrouter-demo")

client = neatlogs.wrap(OpenRouter(api_key=os.environ["OPENROUTER_API_KEY"]))

resp = client.chat.send(
    model="openai/gpt-4o-mini",
    messages=[{"role": "user", "content": "In one sentence, what is OpenRouter?"}],
    temperature=0.3,
    top_p=0.9,
    max_tokens=256,
)
print(resp.choices[0].message.content)

neatlogs.flush()
neatlogs.shutdown()
